### PR TITLE
Course Listing and Course Page - Refactoring and additional page field added

### PR DESCRIPTION
**Pull request by jleong01**
Refactored code to include course listing image asset picker and render content from childpages

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -1,65 +1,63 @@
...
+												jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/select"
+												fieldLabel="Level Of Study" name="./levelOfStudy"
+												rootPath="/content/taylorsuniversity">
+											</levelofstudy>
+											<programme cq:showOnCreate="{Boolean}true"
+												jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/select"
+												fieldLabel="Programme" name="./programme" rootPath="/content/taylorsuniversity" />
+										</items>
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: rootPath for "Select" resourceType?

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -1,65 +1,63 @@
...
+												jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+												fieldLabel="Course Description" name="./course/description"
+												rootPath="/content/taylorsuniversity" />
+											<levelofstudy cq:showOnCreate="{Boolean}true"
+												jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/select"
+												fieldLabel="Level Of Study" name="./levelOfStudy"
+												rootPath="/content/taylorsuniversity">
+											</levelofstudy>
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: rootPath for "Select" resourceType?

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -1,65 +1,63 @@
...
+												text="Change" type="image" />
+											<charset jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/hidden"
+												ignoreData="{Boolean}true" name="_charset_" value="utf-8" />
+											<description cq:showOnCreate="{Boolean}true"
+												jcr:primaryType="nt:unstructured"
+												sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+												fieldLabel="Course Description" name="./course/description"
+												rootPath="/content/taylorsuniversity" />
+											<levelofstudy cq:showOnCreate="{Boolean}true"
```
Comment (bsinghal1) [inline ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: rootPath for "textarea" resourceType?

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml (hunk 1 above)
Comment (kdelosreyes01) [reply ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: removed rootPath

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml (hunk 2 above)
Comment (kdelosreyes01) [reply ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: removed rootPath

File: ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml (hunk 3 above)
Comment (kdelosreyes01) [reply ui.apps/src/main/content/jcr_root/apps/taylorsuniversity/components/page/coursepage/_cq_dialog/.content.xml]: removed rootPath

**Pull request merged**